jina/types/sets/document.py

```diff
@@ -161,35 +171,54 @@ def all_contents(self) -> Tuple['np.ndarray', 'DocumentSet']:
         """
         return self._extract_docs('content')
 
-    def _extract_docs(self, attr: str) -> Tuple['np.ndarray', 'DocumentSet']:
-        contents = []
+    def _extract_docs(
+        self, *fields: str
+    ) -> Tuple[Union['np.ndarray', List['np.ndarray']], 'DocumentSet']:
+        list_of_contents_output = len(fields) > 1
+        contents = [[] for _ in fields if len(fields) > 1]
         docs_pts = []
         bad_docs = []
 
         for doc in self:
-            content = getattr(doc, attr)
+            content = doc.get_attrs_values(*fields)
+            content = content if list_of_contents_output else content[0]
 
             if content is not None:
-                contents.append(content)
+                if list_of_contents_output:
+                    for i, c in enumerate(content):
+                        contents[i].append(c)
+                else:
+                    contents.append(content)
                 docs_pts.append(doc)
             else:
                 bad_docs.append(doc)
 
-        contents = np.stack(contents) if contents else None
+        if list_of_contents_output:
+            for i in range(len(contents)):
+                contents[i] = np.stack(contents[i])
+        else:
+            contents = np.stack(contents) if contents else None
```

Comment:
I would like to refactoring it a bit to make it easier to read. Plus, when `list_of_contents_output==False`, we need to take care the case `doc.get_attrs_values(*fields)` returns an empty list.
```
        if list_of_contents_output:
            for doc in self:
                content = doc.get_attrs_values(*fields)
                if content is None:
                    bad_docs.append(doc)
                    continue
                for i, c in enumerate(content):
                    contents[i].append(c)
                docs_pts.append(doc)
            for i, attr_list in enumerate(contents):
                contents[i] = np.stack(attr_list)
        else:
            for doc in self:
                content = doc.get_attrs_values(*fields)
                if not content:
                    bad_docs.append(doc)
                    continue
                contents.append(content)
                docs_pts.append(doc)
            contents = np.stack(contents) if contents else None
```

Reply:
Done!